libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/SalesforceListActivity.java

```diff
@@ -26,16 +26,17 @@
  */
 package com.salesforce.androidsdk.ui;
 
-import android.app.ListActivity;
 import android.os.Bundle;
 import android.view.KeyEvent;
 
+import androidx.appcompat.app.AppCompatActivity;
+
 /**
  * Abstract base class for all Salesforce list activities.
  *
  * @author bhariharan
  */
-public abstract class SalesforceListActivity extends ListActivity implements SalesforceActivityInterface {
```

Comment:
The abstract classes SalesforceActivity, SalesforceListActivity, SalesforceExpandableListActivity are meant to be extended by customers for their own activity. We should probably leave those alone. We could add new ones (eg extending AppCompatActivity) and use those in our sample apps and templates.

See: https://developer.salesforce.com/docs/atlas.en-us.mobile_sdk.meta/mobile_sdk/android_native_classes_nativemainactivity.htm

Reply:
I have been (sorta) jokingly saying I am going to write a `SalesforceActivity2` for years...